lib/stores/S3Store.ts

```diff
@@ -234,40 +295,58 @@ export default class S3Store extends DataStore {
     readStream: http.IncomingMessage | fs.ReadStream,
     currentPartNumber: number,
     offset: number
-  ): Promise<void> {
+  ): Promise<number> {
     const size = metadata.file.size as number
-    const promises: Promise<void | string>[] = []
+    const promises: Promise<void>[] = []
     let pendingChunkFilepath: string | null = null
+    let bytesUploaded = 0
     const splitterStream = new StreamSplitter({
       chunkSize: this.calcOptimalPartSize(size),
       directory: os.tmpdir(),
     })
       .on('chunkStarted', (filepath) => {
         pendingChunkFilepath = filepath
       })
-      .on('chunkFinished', ({path, size: chunkSize}) => {
+      .on('chunkFinished', ({path, size: partSize}) => {
         pendingChunkFilepath = null
-        offset += chunkSize
-        const partNumber = currentPartNumber++
-        const p = Promise.resolve()
-          .then(() => {
-            // Skip chunk if it is not last and is smaller than 5MB
-            const is_last_chunk = size === offset
-            if (!is_last_chunk && chunkSize < 5 * 1024 * 1024) {
-              log(`[${metadata.file.id}] ignoring chuck smaller than 5MB`)
-              return
+        const deferred = pDefer<void>()
+
+        // Fork off in the background
+        ;(async () => {
```

Comment:
I don't think using a p-defer is useful here?
```suggestion
        const deferred = new Promise(async (resolve, reject) => {
```